presto-main/src/main/java/com/facebook/presto/sql/rewrite/DescribeOutputRewrite.java

```diff
@@ -0,0 +1,166 @@
+import static com.facebook.presto.sql.tree.BooleanLiteral.TRUE_LITERAL;
+import static java.util.Objects.requireNonNull;
+
+final class DescribeOutputRewrite
+        implements StatementRewrite.Rewrite
+{
+    @Override
+    public Statement rewrite(
+            Session session,
+            Metadata metadata,
+            SqlParser parser,
+            Optional<QueryExplainer> queryExplainer,
+            Statement node,
+            List<Expression> parameters,
+            AccessControl accessControl,
+            boolean experimentalSyntaxEnabled)
+    {
+        return (Statement) new Visitor(session, parser, metadata, queryExplainer, parameters, accessControl, experimentalSyntaxEnabled).process(node, null);
+    }
+
+    private static final class Visitor
+            extends AstVisitor<Node, Void>
+    {
+        private final Session session;
+        private final SqlParser parser;
+        private final Metadata metadata;
+        private final Optional<QueryExplainer> queryExplainer;
+        private final List<Expression> parameters;
+        private final AccessControl accessControl;
+        private final boolean experimentalSyntaxEnabled;
+
+        public Visitor(
+                Session session,
+                SqlParser parser,
+                Metadata metadata,
+                Optional<QueryExplainer> queryExplainer,
+                List<Expression> parameters,
+                AccessControl accessControl,
+                boolean experimentalSyntaxEnabled)
+        {
+            this.session = requireNonNull(session, "session is null");
+            this.parser = parser;
+            this.metadata = metadata;
+            this.queryExplainer = queryExplainer;
+            this.parameters = parameters;
+            this.accessControl = accessControl;
+            this.experimentalSyntaxEnabled = experimentalSyntaxEnabled;
+        }
+
+        @Override
+        protected Node visitDescribeOutput(DescribeOutput node, Void context)
+        {
+            String sqlString = session.getPreparedStatement(node.getName());
+            Statement statement = parser.createStatement(sqlString);
+
+            Analyzer analyzer = new Analyzer(session, metadata, parser, accessControl, queryExplainer, experimentalSyntaxEnabled, parameters);
+            Analysis analysis = analyzer.analyze(statement, true);
+
+            Row[] rows = analysis.getRootScope().getRelationType().getVisibleFields().stream().map(field -> createDescribeOutputRow(field, analysis)).toArray(Row[]::new);
+            Query query = simpleQuery(
+                    selectList(nameReference("Column Name"),
+                            nameReference("Table"),
+                            nameReference("Schema"),
+                            nameReference("Connector"),
+                            nameReference("Type"),
+                            nameReference("Type Size"),
+                            nameReference("Aliased"),
```

Comment:
I'm not clear about the semantics of this. What's the expected output if the query has a "join ... using", union/intersect/except, etc., and a column is derived from multiple columns across different tables?